src/fmu/dataio/providers/objectdata/_faultroom.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Final
+
+from fmu.dataio._definitions import ValidFormats
+from fmu.dataio._logging import null_logger
+from fmu.dataio.datastructure.meta import meta, specification
+from fmu.dataio.readers import FaultRoomSurface
+
+from ._base import (
+    DerivedObjectDescriptor,
+    ObjectDataProvider,
+)
+
+logger: Final = null_logger(__name__)
+
+
+@dataclass
+class FaultRoomSurfaceProvider(ObjectDataProvider):
+    obj: FaultRoomSurface
+
+    def get_bbox(self) -> dict[str, Any]:
+        """Derive data.bbox for FaultRoomSurface."""
+        logger.info("Get bbox for FaultRoomSurface")
+        assert isinstance(self.obj, FaultRoomSurface)  # for mypy
+        faultsurf = self.obj
+        return meta.content.BoundingBox3D(
+            xmin=float(faultsurf.bbox["xmin"]),
+            xmax=float(faultsurf.bbox["xmin"]),
+            ymin=float(faultsurf.bbox["ymin"]),
+            ymax=float(faultsurf.bbox["ymax"]),
+            zmin=float(faultsurf.bbox["zmin"]),
+            zmax=float(faultsurf.bbox["zmax"]),
+        ).model_dump(
+            mode="json",
+            exclude_none=True,
+        )
+
+    def get_spec(self) -> dict[str, Any]:
+        """Derive data.spec for FaultRoomSurface"""
+        logger.info("Get spec for FaultRoomSurface")
+        assert isinstance(self.obj, FaultRoomSurface)  # for mypy
```

Comment:
Same here with the assert -- does mypy still complain with the assert removed, but the type hint in the class?